enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,418 @@
+ 
+DriverContainers work on RHCOS/RHEL 7,8, Fedora (on a laptop) and they would work on any other distributions that are able to launch containers, assuming there are builds for such. 
+ 
+With DriverContainers the host stays always "clean", we have no clashes in different library versions or binaries.  Prototyping is far more easier, updates are done by pulling a new container (btw the loading and unloading is done
+by the DriverContainer with checks on /proc and /sys and other files to make sure that all traces are removed).
+
+A detailed description of SRO and its inner workings are described in the following two blog posts:  [https://red.ht/2JQuNwB]https://red.ht/2JQuNwB)
+[https://red.ht/34ubzq3](https://red.ht/34ubzq3). 
+
+We start by describing how our current solutions work and how we combine and simplify them as the proposed solution.
+
+### Special Resource Operator
+ 
+For any day-2 management of kernel modules we can leverage SRO features. SRO was written in  such a way that it is highly customizable and configurable to any hardware accelerator or out-of-tree kernel module. 
+ 
+SRO supports full lifecycle management of an accelerator stack but it can also be used as a stripped down version to manage e.g only one kernel module. Furthermore SRO can handle multiple kernel modules from different vendors and is able to model a dependency between those too.
+ 
+Another important feature is the ability to consume build artifacts from other kernel modules to build a more sophisticated DriverContainer. 
+
+SRO deals with building and managing out-of-tree drivers and vendor stacks that use those drivers. 
+
+Ideally SRO would pull a prebuilt DriverContainer with precompiled drivers from the vendor and just  run it. Updates or downgrades would just be a pull of a container. RHEL provides the DCI (distributed CI) system for customers/partners to get pre-released versions of the kernel with which they could build such DriverContainers. 
+
+We have several commitments from vendors for building such DriverContainers but it is a long way to go. This is where SRO is taking responsibility and filling the gap. 
+The Special Resource Operator (SRO) is currently only available in OperatorHub. SRO has proven in the past to be the template for enabling hardware when on OpenShift. Its capabilities to handle several DriverContainers with only one copy of SRO running makes it a preferable solution to tackle kmods on OpenShift.  
+
+SRO is going to be a core-component of OpenShift and delivered/managed by CVO.  
+
+
+
+### kmods-via-containers (KVC)
+
+kmods-via-containers is a framework for building and delivering kernel modules via containers. This implementation for this framework was inspired by the work done by Joe Doss on atomic-wireguard.
```

Comment:
Links to both projects, please.

Reply:
https://github.com/kmods-via-containers/